src/main/scala/mesosphere/marathon/api/serialization/PortDefinitionSerializer.scala

```diff
@@ -6,18 +6,20 @@ import org.apache.mesos
 import scala.collection.JavaConverters._
 
 object PortDefinitionSerializer {
-  def toProto(portDefinition: PortDefinition): mesos.Protos.Port = {
-    val builder = mesos.Protos.Port.newBuilder
-      .setNumber(portDefinition.port)
-      .setProtocol(portDefinition.protocol)
+  def toProto(portDefinition: PortDefinition): Seq[mesos.Protos.Port] = {
```

Comment:
This method is used for 2 purposes:
- store the port definition as part of the app in zk 
- set discovery information on task

Where the latter now works correctly, you doubled the port information for the first one.

Please introduce a a separate method (e.g. `toMesosProto`) with this implementation and leave the original unchanged. The TaskBuilder has to use `toMesosProto`, the AppDefinitionSerializer `toProto`.